rascal-textmate-core/src/main/rascal/VSCode.rsc

```diff
@@ -23,20 +23,23 @@ RscGrammar getRscGrammar() =
         // The following mapping is based on:
         //   - https://github.com/usethesource/rascal/blob/83023f60a6eb9df7a19ccc7a4194b513ac7b7157/src/org/rascalmpl/values/parsetrees/TreeAdapter.java#L44-L59
         //   - https://github.com/usethesource/rascal-language-servers/blob/752fea3ea09101e5b22ee426b11c5e36db880225/rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/util/SemanticTokenizer.java#L121-L142
+        // With updates based on:
+        //   - https://github.com/eclipse-lsp4j/lsp4j/blob/f235e91fbe2e45f62e185bbb9f6d21bed48eb2b9/org.eclipse.lsp4j/src/main/java/org/eclipse/lsp4j/Protocol.xtend#L5639-L5695
+        //   - https://github.com/usethesource/rascal-language-servers/blob/88be4a326128da8c81d581c2b918b4927f2185be/rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/util/SemanticTokenizer.java#L134-L152
         case \tag("category"("Normal"))           => \tag("category"("source"))
-        case \tag("category"("Type"))             => \tag("category"("storage.type"))
+        case \tag("category"("Type"))             => \tag("category"("type"))     // Updated (before: storage.type)
         case \tag("category"("Identifier"))       => \tag("category"("variable"))
         case \tag("category"("Variable"))         => \tag("category"("variable"))
-        case \tag("category"("Constant"))         => \tag("category"("constant"))
+        case \tag("category"("Constant"))         => \tag("category"("string"))   // Updated (before: constant)
```

Comment:
Making all constant string is not helping. We should do something in the rascal highlighter to fix the underlying issue. Loosing precision here is not so nice.

Reply:
> Making all constant string is not helping

The only productions with the `@category("Constant")` tag in Rascal's own grammar are string-related, so I don't think precision is actually lost here for Rascal's own grammar. (See the other [comment](https://github.com/SWAT-engineering/rascal-textmate/pull/6#issuecomment-2228180407) for a full overview of the usage of `@category` tags.)

So, in the context of Rascal's own grammar, I can understand why the semantic highlighter maps `Constant` to `string`.

> fix the underlying issue

I think the best way forward is to update the categories in Rascal's own grammar (which are "legacy token types") to directly use the intended TextMate scopes. For instance, each occurrence of `@category("Constant")` would then be replaced with `@category("string")` (or, for even more precision, `@category("string.quoted.double")` and `@category("string.quoted.single")`, depending on the production).

However, if it's logistically complicated to update Rascal's own grammar right now, and/or release such an update in the near future, we could also temporarily use a preprocessor to rewrite categories when generating a TextMate grammar -- *only* for Rascal's own grammar (other grammars unaffected). That's essentially this PR.

Reply:
Okay, I assumed numbers would also get a constant assigned? that is weird. We might have to introduce them (maybe via this rewrite logic to add a category for certain productions?)